### ISSUE TITLE
[FEATURE] data source bitwarden_secret - Use key instead of ID

### ISSUE THREAD

**Issue by joryirving** [maxlaverse/terraform-provider-bitwarden#172]
Currently, you can generate and push to a `bitwarden_secret` resource with just a `key`. Obviously this is because the `id` doesn't exist until you create the `key`.

Rather than trying to pass the `id` in to the resource, would we be able to retrieve a bitwarden secret via `key` instead of `id`.

I know previously when BW-SM was added to the `external-secrets` repo, this was requested as a feature, and it was mentioned that you could technically have multiple copies of the same key, so it would be on the user/administrator to make sure this doesn't happen, and if it does, just return the first result.

So it would be something like:

```
data "bitwarden_secret" "example" {
  key = "vpn-secret"
}

resource "kubernetes_secret" "vpn_credentials" {
  metadata {
    name = "vpn-key"
  }

  data = {
    "PASSWORD" = data.bitwarden_secret.value
  }
}
```

Where you would require either `id` or `key`. An added bonus would be if you delete and recreated a secret with the same name, you wouldn't need to update the `id`.

Alternatively maybe a `data.bitwarden_secret_id` resource where you could search by key, and it would return the ID.

```
data "bitwarden_secret_id" "example" {
  key = "vpn-secret"
}

data "bitwarden_secret" "example" {
  id = data.bitwarden_secret_id.example.id
}

resource "kubernetes_secret" "vpn_credentials" {
  metadata {
    name = "vpn-key"
  }

  data = {
    "PASSWORD" = data.bitwarden_secret.value
  }
}
```

**Comment by maxlaverse** [comment]
Agreed, this would make the sourcing of secret much more easier. I would probably prefer to fail if there are multiple results, to avoid debugging confusion and be consistent with the Password Manager data sources.

**Comment by joryirving** [comment]
Does this need a new release to be cut, or will it just work? I haven't had a chance to test it.

**Comment by maxlaverse** [comment]
Yes, it requires a new release. There are a few other changes I'm trying to pack in the next one, but I'll release something soon.

**Comment by maxlaverse** [comment]
Released in [v0.11.0](https://github.com/maxlaverse/terraform-provider-bitwarden/releases/tag/v0.11.0) @joryirving
